test/extended/networking/ovs.go

```diff
@@ -0,0 +1,314 @@
+package networking
+
+import (
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	sdnapi "github.com/openshift/origin/pkg/sdn/api"
+	testexutil "github.com/openshift/origin/test/extended/util"
+	testutil "github.com/openshift/origin/test/util"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kapiunversioned "k8s.io/kubernetes/pkg/api/unversioned"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	hostSubnetTimeout = 30 * time.Second
+)
+
+var _ = Describe("[networking] OVS", func() {
+	Context("generic", func() {
+		f1 := e2e.NewDefaultFramework("net-ovs1")
+		oc := testexutil.NewCLI("get-flows", testexutil.KubeConfigPath())
+
+		It("should add and remove flows when pods are added and removed", func() {
+			nodes := e2e.GetReadySchedulableNodesOrDie(f1.Client)
+			origFlows := getFlowsForAllNodes(oc, nodes.Items)
+			Expect(len(origFlows)).To(Equal(len(nodes.Items)))
+			for _, flows := range origFlows {
+				Expect(len(flows)).ToNot(Equal(0))
+			}
+
+			podName := "ovs-test-webserver"
+			deployNodeName := nodes.Items[0].Name
+			ipPort := e2e.LaunchWebserverPod(f1, podName, deployNodeName)
+			ip := strings.Split(ipPort, ":")[0]
+
+			newFlows := getFlowsForAllNodes(oc, nodes.Items)
+			for _, node := range nodes.Items {
+				if node.Name != deployNodeName {
+					Expect(reflect.DeepEqual(origFlows[node.Name], newFlows[node.Name])).To(BeTrue(), "Flows on non-deployed-to nodes should be unchanged")
+				}
+			}
+
+			var newDeployNodeFlows []string
+			var arpOut, ipOut, arpIn, ipInGeneric, ipInGlobal, ipInIsolated bool
+			for _, flow := range newFlows[deployNodeName] {
+				if strings.Contains(flow, ip) {
+					if strings.Contains(flow, "arp_spa="+ip) {
+						arpOut = true
+					} else if strings.Contains(flow, "arp_tpa="+ip) {
+						arpIn = true
+					} else if strings.Contains(flow, "nw_src="+ip) {
+						ipOut = true
+					} else if strings.Contains(flow, "nw_dst="+ip) {
+						if strings.Contains(flow, "reg0=0x") {
+							ipInIsolated = true
+						} else if strings.Contains(flow, "reg0=0") {
+							ipInGlobal = true
+						} else {
+							ipInGeneric = true
+						}
+					} else {
+						Fail("found unexpected OVS flow: " + flow)
+					}
+				} else {
+					newDeployNodeFlows = append(newDeployNodeFlows, flow)
+				}
+			}
+			Expect(arpOut).To(BeTrue(), "Should have an outgoing ARP rule")
+			Expect(arpIn).To(BeTrue(), "Should have an incoming ARP rule")
+			Expect(ipOut).To(BeTrue(), "Should have an outgoing IP rule")
+			if pluginIsolatesNamespaces() {
+				Expect(ipInGlobal && ipInIsolated).To(BeTrue(), "Should have global and isolated incoming IP rules")
+			} else {
+				Expect(ipInGeneric).To(BeTrue(), "Should have a generic incoming IP rule")
+			}
+			Expect(reflect.DeepEqual(origFlows[deployNodeName], newDeployNodeFlows)).To(BeTrue(), "Deploy node flows should be unchanged except for the new pod")
```

Comment:
Ah, nevermind, I get it now.  newDeployNodeFlows is a confusing name because it doesn't contain all flows, just ones that aren't related to the new pod.